apps/src/templates/teacherDashboard/SectionProgress.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { PropTypes, Component } from 'react';
+import ScriptSelector from './ScriptSelector';
+import { getLevelResult } from '@cdo/apps/code-studio/progressRedux';
+import SectionScriptProgress from './SectionScriptProgress';
+import _ from 'lodash';
+
+/**
+ * Given a particular section, this component owns figuring out which script to
+ * show progress for (selected via a dropdown), and querying the server for
+ * student progress. Child components then have the responsibility for displaying
+ * that progress.
+ */
```

Comment:
Decided to separate the data gathering component from the UI component for the most part. That said, this component does still include the dropdown.